react/src/screens/SignUp/index.js

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+
+import routes from '../../constants/routes';
+import {validateEmail, validatePasswordLength, validatePasswordContent, validateNameContent, validateSurnameContent} from '../../utils/validations';
+import InputWideWithHeader from '../../components/InputWideWithHeader';
+
+import './styles.css';
+import strings from './strings';
+
+class SignUp extends Component {
+  state = {email: '', password: '', name: '', surname: '', errorEmail: '', errorPassword: '', errorName: '', errorSurname: ''};
+
+  submitHandler = e => {
+    let errorEmail, errorPassword, errorName, errorSurname = "";
+    errorEmail = validateEmail(this.state.email);
+    errorPassword = validatePasswordLength(this.state.password) || validatePasswordContent(this.state.password);
+    errorName = validateNameContent(this.state.name);
+    errorSurname = validateSurnameContent(this.state.surname);
+
+    if(errorEmail || errorPassword || errorName || errorSurname){
+      errorEmail = errorEmail || '';
+      errorPassword = errorPassword || '';
+      errorName = errorName || '';
+      errorSurname = errorSurname || '';
+      this.setState({errorEmail, errorPassword, errorName, errorSurname});
+      e.preventDefault();
+    }
+  }
+
+  handleEmailChange = (e) => { this.setState({email: e.target.value}); }
+  handlePasswordChange = (e) => { this.setState({password: e.target.value}); }
+  handleNameChange = (e) => { this.setState({name: e.target.value}); }
+  handleSurnameChange = (e) => { this.setState({surname: e.target.value}); }
```

Comment:
This logic is also start to being repeated. Would you like to extract a function from these ? 😇